pkg/apis/authentication/types.go

```diff
@@ -162,3 +162,24 @@ type BoundObjectReference struct {
 	// UID of the referent.
 	UID types.UID
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SelfSubjectAttributesReview is a request to authenticate in kubernetes api
+// and self attributes returned by authenticator.
+type SelfSubjectAttributesReview struct {
+	metav1.TypeMeta
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta
+	// Status is filled in by the server with the user attributes.
+	Status SelfSubjectAttributesReviewStatus
+}
+
+// SelfSubjectAttributesReviewStatus is filled by the kube-apiserver and sent back to a user.
+type SelfSubjectAttributesReviewStatus struct {
+	// User attributes of the current user.
+	// +optional
+	UserInfo UserInfo
```

Comment:
note to self, this reuse looks correct because the type is intended to implement the user.Info interface.